### PR TITLE
Remove some usages of RskSystemProperties.CONFIG

### PR DESCRIPTION
**Pull request by tinchou**
Here's some help to tackle this PR:

1. every commit is one refactor
1. I reviewed my changes before every commit, so I hopefully caught all issues
1. I used IntelliJ's automatic refactoring tools, so I'm pretty confident about my changes
1. I focused on places where it was easy to inject `RskSystemProperties.CONFIG`, but there's lots of work left to be done
1. I usually pushed `RskSystemProperties.CONFIG` all the way up to test code, where we don't really mind this being used
1. our final goal is to remove this global variable from production code, but it wouldn't be bad to have it in test code

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 392 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 172 major
* ![MINOR][MINOR] 128 minor
* ![INFO][INFO] 89 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [MinerClientImpl.java#L116](https://github.com/rsksmart/rskj/blob/5a1b55b3b8aebac9bcf625ec3e3181079b963173/rskj-core/src/main/java/co/rsk/mine/MinerClientImpl.java#L116): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [MinerClientImpl.java#L124](https://github.com/rsksmart/rskj/blob/5a1b55b3b8aebac9bcf625ec3e3181079b963173/rskj-core/src/main/java/co/rsk/mine/MinerClientImpl.java#L124): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [MinerClientImpl.java#L136](https://github.com/rsksmart/rskj/blob/5a1b55b3b8aebac9bcf625ec3e3181079b963173/rskj-core/src/main/java/co/rsk/mine/MinerClientImpl.java#L136): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![MAJOR][MAJOR] [Start.java#L65](https://github.com/rsksmart/rskj/blob/5a1b55b3b8aebac9bcf625ec3e3181079b963173/rskj-core/src/main/java/co/rsk/Start.java#L65): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L83](https://github.com/rsksmart/rskj/blob/5a1b55b3b8aebac9bcf625ec3e3181079b963173/rskj-core/src/main/java/co/rsk/Start.java#L83): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskFactory.java#L90](https://github.com/rsksmart/rskj/blob/5a1b55b3b8aebac9bcf625ec3e3181079b963173/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L90): Method has 14 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskFactory.java#L186](https://github.com/rsksmart/rskj/blob/5a1b55b3b8aebac9bcf625ec3e3181079b963173/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L186): This block of commented-out lines of code should be removed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [RskFactory.java#L230](https://github.com/rsksmart/rskj/blob/5a1b55b3b8aebac9bcf625ec3e3181079b963173/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L230): Method has 13 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L101](https://github.com/rsksmart/rskj/blob/5a1b55b3b8aebac9bcf625ec3e3181079b963173/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L101): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L118](https://github.com/rsksmart/rskj/blob/5a1b55b3b8aebac9bcf625ec3e3181079b963173/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L118): Constructor new co.rsk.core.bc.BlockChainImpl(Repository, BlockStore, ReceiptStore, PendingState, EthereumListener, AdminInfo, BlockValidator, RskSystemProperties) makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java [hunk 1]
```diff
@@ -159,35 +160,40 @@ public RskSystemProperties rskSystemProperties() {
...
-        BlockStore blockStore = appCtx.getBean(BlockStore.class);
-        int uncleListLimit = RskSystemProperties.CONFIG.getBlockchainConfig().getCommonConstants().getUncleListLimit();
-        int uncleGenLimit = RskSystemProperties.CONFIG.getBlockchainConfig().getCommonConstants().getUncleGenerationLimit();
-        int validPeriod = RskSystemProperties.CONFIG.getBlockchainConfig().getCommonConstants().getNewBlockMaxMinInTheFuture();
+    public BlockValidationRule blockValidationRule(
+            BlockStore blockStore,
+            RskSystemProperties config,
+            DifficultyCalculator difficultyCalculator,
+            ProofOfWorkRule proofOfWorkRule) {
+        int uncleListLimit = config.getBlockchainConfig().getCommonConstants().getUncleListLimit();
```
Comment (lsebrie) [inline rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java]: add a var with the value of config.getBlockchainConfig().getCommonConstants() to make it a bit cleaner

File: rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java [hunk 2]
```diff
@@ -200,24 +206,26 @@ public NetworkStateExporter networkStateExporter() {
...
-        BlockParentGasLimitRule parentGasLimitRule = new BlockParentGasLimitRule(RskSystemProperties.CONFIG.getBlockchainConfig().getCommonConstants().getGasLimitBoundDivisor());
-        BlockParentCompositeRule unclesBlockParentHeaderValidator = new BlockParentCompositeRule(new PrevMinGasPriceRule(), new BlockParentNumberRule(), new BlockDifficultyRule(), parentGasLimitRule);
-
-        int validPeriod = RskSystemProperties.CONFIG.getBlockchainConfig().getCommonConstants().getNewBlockMaxMinInTheFuture();
+    public BlockValidationRule minerServerBlockValidationRule(
+            BlockStore blockStore,
+            RskSystemProperties config,
+            DifficultyCalculator difficultyCalculator,
+            ProofOfWorkRule proofOfWorkRule) {
+        int uncleListLimit = config.getBlockchainConfig().getCommonConstants().getUncleListLimit();
```
Comment (lsebrie) [inline rskj-core/src/main/java/org/ethereum/config/DefaultConfig.java]: Same as BlockValidationRule bean goes here, make a var with commonConstants

File: rskj-core/src/test/java/co/rsk/blockchain/utils/BlockGenerator.java [hunk 3]
```diff
@@ -75,7 +77,7 @@ public static BlockGenerator getInstance() {
         return INSTANCE;
     }
 
-
+    private final DifficultyCalculator difficultyCalculator = new DifficultyCalculator(RskSystemProperties.CONFIG);
```
Comment (diega) [inline rskj-core/src/test/java/co/rsk/blockchain/utils/BlockGenerator.java]: Can this field be passed in the constructor? Also changing the deprecated *INSTANCE* to
```java
 private static final BlockGenerator INSTANCE = new BlockGenerator(new DifficultyCalculator(RskSystemProperties.CONFIG));
```

File: rskj-core/src/main/java/org/ethereum/facade/EthereumFactory.java [hunk 4]
```diff
@@ -19,7 +19,6 @@
```
Comment (diega) [inline rskj-core/src/main/java/org/ethereum/facade/EthereumFactory.java]: This Class and `org.ethereum.Start` can be safely removed, they were part of how _EthereumJ_ was intialized, and also it's dangerous to have another class than `co.rsk.Start` creating an Application Context

File: rskj-core/src/main/java/co/rsk/mine/MinerClientImpl.java [hunk 5]
```diff
@@ -21,7 +21,6 @@
 import co.rsk.config.RskSystemProperties;
 import co.rsk.core.Rsk;
 import co.rsk.panic.PanicProcessor;
-import com.google.common.annotations.VisibleForTesting;
```
Comment (diega) [inline rskj-core/src/main/java/co/rsk/mine/MinerClientImpl.java]: God bless you

File: rskj-core/src/main/java/co/rsk/core/RskFactory.java [hunk 6]
```diff
@@ -200,15 +201,16 @@ public NodeMessageHandler getNodeMessageHandler(NodeBlockProcessor nodeBlockProc
                                                     ChannelManager channelManager,
                                                     PendingState pendingState,
                                                     TxHandler txHandler,
-                                                    PeerScoringManager peerScoringManager) {
+                                                    PeerScoringManager peerScoringManager,
+                                                    ProofOfWorkRule ProofOfWorkRule) {
```
Comment (diega) [inline rskj-core/src/main/java/co/rsk/core/RskFactory.java]: Please, rename to `proofOfWorkRule`

File: rskj-core/src/test/java/co/rsk/blockchain/utils/BlockGenerator.java (hunk 3 above)
Comment (tinchou) [reply rskj-core/src/test/java/co/rsk/blockchain/utils/BlockGenerator.java]: This is test code and that change would introduce more changes than I want. Let's leave this to the person in the future that needs to inject a different DifficultyCalculator instance.

**Comment by rskops** [comment]
SonarQube analysis reported 384 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 168 major
* ![MINOR][MINOR] 125 minor
* ![INFO][INFO] 88 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [MinerClientImpl.java#L116](https://github.com/rsksmart/rskj/blob/9fbbc7acd2df555fd2e022c161466b6ec443d60b/rskj-core/src/main/java/co/rsk/mine/MinerClientImpl.java#L116): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [MinerClientImpl.java#L124](https://github.com/rsksmart/rskj/blob/9fbbc7acd2df555fd2e022c161466b6ec443d60b/rskj-core/src/main/java/co/rsk/mine/MinerClientImpl.java#L124): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![CRITICAL][CRITICAL] [MinerClientImpl.java#L136](https://github.com/rsksmart/rskj/blob/9fbbc7acd2df555fd2e022c161466b6ec443d60b/rskj-core/src/main/java/co/rsk/mine/MinerClientImpl.java#L136): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2142)
1. ![MAJOR][MAJOR] [Start.java#L65](https://github.com/rsksmart/rskj/blob/9fbbc7acd2df555fd2e022c161466b6ec443d60b/rskj-core/src/main/java/co/rsk/Start.java#L65): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L83](https://github.com/rsksmart/rskj/blob/9fbbc7acd2df555fd2e022c161466b6ec443d60b/rskj-core/src/main/java/co/rsk/Start.java#L83): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskFactory.java#L90](https://github.com/rsksmart/rskj/blob/9fbbc7acd2df555fd2e022c161466b6ec443d60b/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L90): Method has 14 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [RskFactory.java#L186](https://github.com/rsksmart/rskj/blob/9fbbc7acd2df555fd2e022c161466b6ec443d60b/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L186): This block of commented-out lines of code should be removed. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [RskFactory.java#L230](https://github.com/rsksmart/rskj/blob/9fbbc7acd2df555fd2e022c161466b6ec443d60b/rskj-core/src/main/java/co/rsk/core/RskFactory.java#L230): Method has 13 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L101](https://github.com/rsksmart/rskj/blob/9fbbc7acd2df555fd2e022c161466b6ec443d60b/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L101): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [BlockChainImpl.java#L118](https://github.com/rsksmart/rskj/blob/9fbbc7acd2df555fd2e022c161466b6ec443d60b/rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java#L118): Constructor new co.rsk.core.bc.BlockChainImpl(Repository, BlockStore, ReceiptStore, PendingState, EthereumListener, AdminInfo, BlockValidator, RskSystemProperties) makes call to non-final method [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APCOA_PARTIALLY_CONSTRUCTED_OBJECT_ACCESS)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**